pkg/secrethub/client.go

```diff
@@ -42,6 +46,27 @@ type Client struct {
 	// repoindexKeys are the keys used to generate blind names in the repo.
 	// These are cached
 	repoIndexKeys map[api.RepoPath]*crypto.SymmetricKey
+
+	appInfo *AppInfo
+}
+
+// AppInfo contains information about the application that is using the SecretHub client.
+// It is used to identify the application to the SecretHub API.
+type AppInfo struct {
+	Name            string
+	Version         string
+	OperatingSystem string
```

Comment:
Do we want to add a `URL` field with a url of the project? Would help us a bunch too I guess.

Reply:
Also, do we want to add a utility function or something to help auto-detect the OS? And, what info do we want to include in the OS? Found this library that can do quite a lot it seems: https://github.com/matishsiao/goInfo

Reply:
Or even better yet, the Docker team has invented this wheel already:

- Detect OS: https://github.com/moby/moby/blob/master/pkg/parsers/operatingsystem/operatingsystem_linux.go
- Detect Kernel: https://github.com/moby/moby/tree/master/pkg/parsers/kernel
- Pack version information into UserAgent header format: https://github.com/moby/moby/tree/master/pkg/useragent